stdlib/forwardable.rb

```diff
@@ -1,16 +1,30 @@
-module Forwardable
-  def instance_delegate(hash)
-    hash.each {|methods, accessor|
-      methods = [methods] unless methods.respond_to? :each
-
-      methods.each {|method|
-        def_instance_delegator(accessor, method)
+%x{
+  function forwardToDelegator(hash, delegator) {
+    for (var i = 0, keys = hash.$$keys, smap = hash.$$smap, len = keys.length, key, value; i < len; i++) {
+      methods = keys[i];
+      if (key.$$is_string) {
+        accessor = smap[key];
+      } else {
+        accessor = key.value;
+        methods = key.key;
+      }
+      if (!methods['$respond_to?']("each")) {
+        methods = [methods];
+      }
+      for (var j = 0; j < methods.length; j++) {
+        delegator(accessor, methods[j]);
       }
     }
+  }
+}
+
+module Forwardable
+  def instance_delegate(hash)
+    `forwardToDelegator(hash, def_instance_delegator)`
```

Comment:
@iliabylich The code just feels wrong but somehow the tests suite passed... function `def_instance_delegator` should be undefined in the JavaScript environment ?

Reply:
Initially I thought that it could be possible because we compile `def def_instance_delegator` to `Opal.def(self, function def_instance_delegator ...` and so the function becomes accessible because of hoisting. But we have a `$$` prefix for generated function names, so it shouldn't work. 

The reason whey the build is green is simple: there are no tests for `Forwardable` in the rubyspec test suite.